pycbc/waveform/ringdown.py

```diff
@@ -457,23 +457,28 @@ def fd_damped_sinusoid(f_0, tau, amp, phi, delta_f, f_lower, f_final, t_0=0.,
 #### Base multi-mode for all approximants
 ######################################################
 
-def multimode_base(input_params):
+def multimode_base(input_params, fd_approximant=False, td_approximant=False,
+                   freq_tau_approximant=False):
     """Return a superposition of damped sinusoids in either time or frequency
     domains with parameters set by input_params.
     """
     input_params['lmns'] = format_lmns(input_params['lmns'])
     amps, phis = lm_amps_phases(**input_params)
-    if 'final_mass' in input_params.keys():
+    if freq_tau_approximant or 'final_mass' not in input_params.keys():
+        freqs, taus = lm_freqs_taus(**input_params)
+        norm = 1.
+    else:
         freqs, taus = get_lm_f0tau_allmodes(input_params['final_mass'],
                         input_params['final_spin'], input_params['lmns'])
         norm = Kerr_factor(input_params['final_mass'],
             input_params['distance']) if 'distance' in input_params.keys() \
             else 1.
-    else:
-        freqs, taus = lm_freqs_taus(**input_params)
-        norm = 1.
-
-    if 'delta_t' in input_params.keys():
+    if not (fd_approximant or td_approximant):
+        if 'delta_t' in input_params.keys():
+            td_approximant = True
+        elif 'delta_f' in input_params.keys():
+            fd_approximant = True
```

Comment:
Get rid of this if block when you change to a single `domain` argument; then just evaluate the `domain` below, raising an error if it isn't either `fd` or `td`. That means that you won't be trying to guess what to use based on the input params, but I think that's for the best. Having it try to guess can lead to unexpected results.